File: spectral_library/lib/Models/folder.dart
import 'package:spectral_library/Models/spect_file.dart';

class Folder {
  int? folderId;
  String folderName;
  List<SpectFile>? files;
  Folder({this.folderId, required this.folderName, this.files});
  Map<String, dynamic> toMap() {
    return {
      'folderId': folderId,
      'folderName': folderName,
      'files': files?.map((file) => file.toMap()).toList(),
    };
  }

  /// Creates a Folder object from a Map
  factory Folder.fromMap(Map<String, dynamic> map) {
    return Folder(
      folderId: map['folderId'] as int?,
      folderName: map['folderName'] as String,
      files: (map['files'] as List<dynamic>)
          .map((fileMap) => SpectFile.fromMap(fileMap as Map<String, dynamic>))
          .toList(),
    );
  }
}

File: backend/Models/Folder.py
from typing import Optional, List

from Models.SpectFile import SpectFile

class Folder:
    def __init__(
        self,
        folder_name: str,
        files: List[SpectFile],
        folder_id: Optional[int] = None,
    ):
        self.folder_id = folder_id
        self.folder_name = folder_name
        self.files = files

    def to_map(self):
        return {
            "folderId": self.folder_id,
            "folderName": self.folder_name,
            "files": [file.to_map() for file in self.files],
        }
        
    @classmethod
    def from_map(cls, data):
        return cls(
            folder_id=data.get("folderId"),
            folder_name=data["folderName"],
            files=[SpectFile.from_map(file) for file in (data["files"] or [])],
        )